core/services/functions/listener.go

```diff
@@ -670,7 +797,61 @@ func (l *FunctionsListener) reportSourceCodeDomains(requestId RequestID, domains
 	}
 }
 
-func (l *FunctionsListener) getSecrets(ctx context.Context, eaClient ExternalAdapterClient, requestID string, subscriptionOwner common.Address, requestData *RequestData) (decryptedSecrets string, userError, internalError error) {
+func (l *FunctionsListener) getSecretsV1(ctx context.Context, eaClient ExternalAdapterClient, requestID string, subscriptionOwner common.Address, requestData *RequestDataV1) (decryptedSecrets string, userError, internalError error) {
```

Comment:
Since the only requestData fields we need for getSecrets are SecretsLocation and Secrets, we could create a new interface which contains those fields and just cast RequestDataV0 or RequestDataV1 to that interface.